lib/mayaUsd/fileio/translators/translatorMayaReference.cpp

```diff
@@ -424,29 +475,74 @@ MStatus UsdMayaTranslatorMayaReference::update(const UsdPrim& prim, MObject pare
-                //
-                if (refName == expectedRefName) {
-                    // Reconnect the reference node's `associatedNode` attr before
-                    // loading it, since the previous connection may be gone.
-                    connectReferenceAssociatedNode(parentDag, tempRefFn);
-                    refNode = tempRefNode;
-                    break;
+        if (isMayaReference(prim)) {
+            // New behaviour
+            if (prim.HasAttribute(MayaReferenceAttribute)) {
+                UsdAttribute attr = prim.GetAttribute(MayaReferenceAttribute);
+                VtValue      value;
+                attr.Get(&value);
+                auto resInfo = attr.GetResolveInfo();
+                // Check if attribute was directly on prim or inherited.
+                bool isAttributeOnPrim = resInfo.GetNode().GetPath() == prim.GetPath();
+                if (isAttributeOnPrim && value.GetType() == SdfValueTypeNames->String.GetType()) {
+                    MString expectedValue = MString(value.Get<std::string>().c_str());
+                    for (MItDependencyNodes refIter(MFn::kReference); !refIter.isDone();
+                         refIter.next()) {
+                        MObject      tempRefNode = refIter.item();
+                        MFnReference tempRefFn(tempRefNode);
+                        if (!tempRefFn.isFromReferencedFile()) {
+                            if (expectedValue == tempRefFn.name()) {
+                                // Reconnect the reference node's `associatedNode` attr before
+                                // loading it, since the previous connection may be gone.
+                                connectReferenceAssociatedNode(parentDag, tempRefFn);
+                                refNode = tempRefNode;
+
+                                // Update Maya Ref Node on reconnect
+                                MString uniqueRefNodeName
+                                    = getUniqueRefNodeName(prim, parentDag, tempRefFn);
+                                tempRefFn.setName(uniqueRefNodeName);
+
+                                MString refDependNodeName = tempRefFn.name();
+                                VtValue value(std::string(
+                                    refDependNodeName.asChar(), refDependNodeName.length()));
+                                prim.SetCustomDataByKey(maya_associatedReferenceNode, value);
+
+                                if (prim.HasAttribute(MayaReferenceAttribute)) {
+                                    UsdAttribute attr = prim.GetAttribute(MayaReferenceAttribute);
+                                    attr.Set(refDependNodeName.asChar());
+                                }
+                                break;
+                            }
+                        }
+                    }
+                }
+            }
+        } else {
+            // Old behaviour
```

Comment:
Old behavior seems a mis-nomer: it's current behavior for non-Maya references, right?